### DISCUSSION TITLE
Deprecated code in 0x06d-Testing-Data-Storage.md (synchronize)

### DISCUSSION THREAD

**Discussion by vtourraine** [OWASP/mastg#2324] [General]
Hello. The sample code in https://github.com/OWASP/owasp-mastg/blob/master/Document/0x06d-Testing-Data-Storage.md#keychain-data-persistence includes a method that Apple says “shouldn’t be used”.

Here’s the line in question:

``` swift
userDefaults.synchronize() // Forces the app to update UserDefaults
```

And the official documentation for the method: https://developer.apple.com/documentation/foundation/nsuserdefaults/1414005-synchronize

> this method is unnecessary and shouldn't be used

I think we should update the sample code to remove this line. Should I open an issue?

**Comment by cpholguera** [comment]
Hi @vtourraine, it would be great if you could open a pull request with the change. In this case it's very clear what's to be done.

Would you be able to test the change using an app? Just to ensure that it really makes no difference.

Thanks a lot for reporting this!

**Reply by vtourraine** [reply]
Thank you for your feedback 👍 

I’ve just submitted a pull request: #2325 

> Would you be able to test the change using an app? Just to ensure that it really makes no difference.

No, I don’t see how to test that, and unfortunately this code isn’t open sourced by Apple. Someone might be able to decompile the SDK and look for the raw implementation, but I’m afraid that’s beyond my skillset.
